Dockerfile-windows.template

```diff
@@ -0,0 +1,31 @@
+FROM mcr.microsoft.com/windows/servercore:%%WINDOWS-VERSION%%
+
+SHELL ["cmd", "/S", "/C"]
+
+ENV RUST_VERSION=%%RUST-VERSION%%
+
+# Install the Visual Studio Build Tools.
+ADD https://aka.ms/vs/16/release/vs_buildtools.exe /vs_buildtools.exe
+RUN vs_buildtools.exe --quiet --wait --norestart --nocache \
+        --installPath C:\BuildTools \
+        --add Microsoft.Component.MSBuild \
+        --add Microsoft.VisualStudio.Component.Windows10SDK.%%SDK-BUILD%% \
+        --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 \
+    || IF "%ERRORLEVEL%"=="3010" EXIT 0
+RUN del vs_buildtools.exe
+RUN setx /M __VSCMD_ARG_NO_LOGO "1"
+
+ENTRYPOINT ["powershell.exe", "-NoLogo", "-ExecutionPolicy", "Bypass"]
```

Comment:
This block of steps should be merged into a single one to keep the total image size down.